File: lista_1/app/src/main/java/com/example/quizapplication/MainActivity.java
package com.example.quizapplication;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ProgressBar;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private TextView questionText, questionCounter, result;
    private RadioGroup radioGroup;
    private RadioButton option1, option2, option3, option4;
    private Button nextButton;
    private ProgressBar progressBar;

    private final String[] questions = {
            "Jednostką siły w układzie SI jest:",
            "Wzór na prędkość w ruchu jednostajnym prostoliniowym to:",
            "Czym jest praca w fizyce?",
            "Jakie jest przyspieszenie ziemskie na powierzchni Ziemi?",
            "Co to jest pętla w programowaniu?",
            "Co oznacza skrót RAM?",
            "Która z poniższych jednostek jest największa?",
            "Który z poniższych języków programowania jest językiem niskiego poziomu?",
            "W jakim systemie liczbowym działają komputery?",
            "Co oznacza skrót HTML?"
    };

    private final String[][] answers = {
            {"Kilogram", "Niuton", "Dżul", "Wat"},
            {"v = a ⋅ t", "v = s / t", "v = m ⋅ g", "v = t / s"},
            {"Wynikiem siły działającej na ciało na odcinku drogi", "Objętością ciała", "Ilością ciepła w procesie", "Czasem trwania ruchu"},
            {"8,9 m/s²", "10,8 m/s²", "9,8 m/s²", "11,8 m/s²"},
            {"Funkcja do przechowywania danych", "Instrukcja umożliwiająca wielokrotne wykonanie bloku kodu", "Typ danych", "Kod źródłowm programu"},
            {"Read-Only Memory", "Random Access Memory", "Random Action Module", "Remote Access Memory"},
            {"Gigabajt", "Megabajt", "Bajt", "Kilobajt"},
            {"Python", "C++", "Java", "Assembler"},
            {"Dziesiętnym", "Binarnym", "Ósemkowym", "Szesnastkowym"},
            {"Hyperlink and Text Markup Language", "Hyper Transfer Markup Language", "Hyper Text Markup Language", "High Text Markup Language"}
    };

    private final int[] correctAnswers = {1, 1, 0, 2, 1, 1, 0, 3, 1, 2};
    private int currentQuestionIndex = 0;
    private int score = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        questionText = findViewById(R.id.question);
        questionCounter = findViewById(R.id.question_counter);
        result = findViewById(R.id.result);
        radioGroup = findViewById(R.id.radio_group);
        option1 = findViewById(R.id.button_1);
        option2 = findViewById(R.id.button_2);
        option3 = findViewById(R.id.button_3);
        option4 = findViewById(R.id.button_4);
        nextButton = findViewById(R.id.button_next);
        progressBar = findViewById(R.id.progress_bar);

        displayQuestion();

        nextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int selectedOptionId = radioGroup.getCheckedRadioButtonId();
                if (selectedOptionId == -1) {
                    return;
                }

                int selectedAnswerIndex = radioGroup.indexOfChild(findViewById(selectedOptionId));

                if (selectedAnswerIndex == correctAnswers[currentQuestionIndex]) {
                    score++;
                }

                currentQuestionIndex++;
                if (currentQuestionIndex < questions.length) {
                    displayQuestion();
                }
                else {
                    showResult();
                }
            }
        });
    }

    private void displayQuestion() {
        questionText.setText(questions[currentQuestionIndex]);
        option1.setText(answers[currentQuestionIndex][0]);
        option2.setText(answers[currentQuestionIndex][1]);
        option3.setText(answers[currentQuestionIndex][2]);
        option4.setText(answers[currentQuestionIndex][3]);

        questionCounter.setText("Pytanie: " + (currentQuestionIndex + 1) + "/" + questions.length);

        radioGroup.clearCheck();

        progressBar.setProgress((currentQuestionIndex) * 10);
    }

    private void showResult() {
        questionText.setVisibility(View.INVISIBLE);
        radioGroup.setVisibility(View.INVISIBLE);
        progressBar.setVisibility(View.INVISIBLE);

        questionCounter.setText("Gratulacje");

        result.setText("Twój wynik: " + score + "/10");
        result.setVisibility(View.VISIBLE);

        nextButton.setText("Restart");
        nextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                resetQuiz();
            }
        });
    }

    private void resetQuiz() {
        currentQuestionIndex = 0;
        score = 0;

        questionText.setVisibility(View.VISIBLE);
        radioGroup.setVisibility(View.VISIBLE);
        progressBar.setVisibility(View.VISIBLE);
        questionCounter.setVisibility(View.VISIBLE);
        result.setVisibility(View.INVISIBLE);

        nextButton.setText("Następne");

        displayQuestion();

        nextButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int selectedOptionId = radioGroup.getCheckedRadioButtonId();
                if (selectedOptionId == -1) {
                    return;
                }

                int selectedAnswerIndex = radioGroup.indexOfChild(findViewById(selectedOptionId));
                if (selectedAnswerIndex == correctAnswers[currentQuestionIndex]) {
                    score++;
                }

                currentQuestionIndex++;
                if (currentQuestionIndex < questions.length) {
                    displayQuestion();
                }
                else {
                    showResult();
                }
            }
        });
    }
}

File: lista_4/app/src/main/java/com/example/lista_4/MainActivity.kt
package com.example.lista_4

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

class MainActivity : ComponentActivity() {

    private val questions = listOf(
        "Jednostką siły w układzie SI jest:",
        "Wzór na prędkość w ruchu jednostajnym prostoliniowym to:",
        "Czym jest praca w fizyce?",
        "Jakie jest przyspieszenie ziemskie na powierzchni Ziemi?",
        "Co to jest pętla w programowaniu?",
        "Co oznacza skrót RAM?",
        "Która z poniższych jednostek jest największa?",
        "Który z poniższych języków programowania jest językiem najniższego poziomu?",
        "W jakim systemie liczbowym działają komputery?",
        "Co oznacza skrót HTML?"
    )

    private val answers = listOf(
        listOf("Kilogram", "Niuton", "Dżul", "Wat"),
        listOf("v = a ⋅ t", "v = s / t", "v = m ⋅ g", "v = t / s"),
        listOf("Wynikiem siły działającej na ciało na odcinku drogi", "Objętością ciała", "Ilością ciepła w procesie", "Czasem trwania ruchu"),
        listOf("8,9 m/s²", "10,8 m/s²", "9,8 m/s²", "11,8 m/s²"),
        listOf("Funkcja do przechowywania danych", "Instrukcja umożliwiająca wielokrotne wykonanie bloku kodu", "Typ danych", "Kod źródłowy programu"),
        listOf("Read-Only Memory", "Random Access Memory", "Random Action Module", "Remote Access Memory"),
        listOf("Gigabajt", "Megabajt", "Bajt", "Kilobajt"),
        listOf("Python", "C++", "Java", "Assembler"),
        listOf("Dziesiętnym", "Binarnym", "Ósemkowym", "Szesnastkowym"),
        listOf("Hyperlink and Text Markup Language", "Hyper Transfer Markup Language", "Hyper Text Markup Language", "High Text Markup Language")
    )

    private val correctAnswers = listOf(1, 1, 0, 2, 1, 1, 0, 3, 1, 2)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            QuizApp(questions, answers, correctAnswers)
        }
    }
}

@Composable
fun QuizApp(questions: List<String>, answers: List<List<String>>, correctAnswers: List<Int>) {
    var currentQuestionIndex by remember { mutableStateOf(0) }
    var score by remember { mutableStateOf(0) }
    var selectedOption by remember { mutableStateOf<Int?>(null) }
    var showResult by remember { mutableStateOf(false) }

    if (showResult) {
        ResultScreen(score = score, totalQuestions = questions.size) {
            currentQuestionIndex = 0
            score = 0
            selectedOption = null
            showResult = false
        }
    } else {
        QuestionScreen(
            question = questions[currentQuestionIndex],
            answers = answers[currentQuestionIndex],
            questionIndex = currentQuestionIndex,
            totalQuestions = questions.size,
            selectedOption = selectedOption,
            onOptionSelected = { selectedOption = it },
            onNextClicked = {
                if (selectedOption == correctAnswers[currentQuestionIndex]) {
                    score++
                }
                if (currentQuestionIndex + 1 < questions.size) {
                    currentQuestionIndex++
                    selectedOption = null
                } else {
                    showResult = true
                }
            }
        )
    }
}

@Composable
fun QuestionScreen(
    question: String,
    answers: List<String>,
    questionIndex: Int,
    totalQuestions: Int,
    selectedOption: Int?,
    onOptionSelected: (Int) -> Unit,
    onNextClicked: () -> Unit
) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.Start
    ) {
        Text(
            text = "Pytanie ${questionIndex + 1}/$totalQuestions",
            textAlign = TextAlign.Center,
            fontSize = 30.sp,
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 16.dp)
        )
        Spacer(modifier = Modifier.height(30.dp))

        LinearProgressIndicator(
            progress = {(questionIndex + 1).toFloat() / totalQuestions.toFloat()},
            modifier = Modifier
                .fillMaxWidth()
        )
        Spacer(modifier = Modifier.height(30.dp))

        Box(
            modifier = Modifier
                .fillMaxWidth()
                .background(Color.LightGray, RoundedCornerShape(8.dp))
                .padding(16.dp)
        ) {
            Text(
                text = question,
                fontSize = 25.sp,
            )
        }
        Spacer(modifier = Modifier.height(30.dp))

        answers.forEachIndexed { index, answer ->
            Row(
                verticalAlignment = Alignment.CenterVertically,
                modifier = Modifier
                    .fillMaxWidth()
                    .background(Color.LightGray, RoundedCornerShape(8.dp))
            ) {
                RadioButton(
                    selected = selectedOption == index,
                    onClick = { onOptionSelected(index) },
                )
                Text(
                    text = answer,
                    fontSize = 20.sp,
                )
            }
            Spacer(modifier = Modifier.height(16.dp))
        }
        Spacer(modifier = Modifier.height(8.dp))

        Button(
            onClick = onNextClicked,
            enabled = selectedOption != null,
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 16.dp)
        ) {
            Text(
                text = "Następne",
                fontSize = 20.sp
            )
        }
    }
}

@Composable
fun ResultScreen(score: Int, totalQuestions: Int, onRestart: () -> Unit) {
    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(16.dp),
        verticalArrangement = Arrangement.Center,
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = "Gratulacje!",
            fontSize = 28.sp,
            modifier = Modifier.padding(bottom = 16.dp)
        )
        Text(
            text = "Twój wynik: $score/$totalQuestions",
            fontSize = 24.sp,
            modifier = Modifier.padding(16.dp)
        )
        Spacer(modifier = Modifier.height(16.dp))
        Button(onClick = onRestart) {
            Text(text = "Zrestartuj quiz")
        }
    }
}
